packages/legend-query/src/stores/QueryBuilderExplorerState.ts

```diff
@@ -282,12 +283,33 @@ const isAutoMappedProperty = (
   return false;
 };
 
+export const generatePropertyNodeMappingDataForNonPrimitveProperty = (
+  propertyMapping: PropertyMapping,
+): QueryBuilderPropertyMappingData => {
+  const targetSetImpl =
+    resolveTargetSetImplementationForPropertyMapping(propertyMapping);
+  return {
+    mapped: true,
+    // NOTE: we could potentially resolve all the leaves and then overlap them somehow to
+    // help identifying the mapped properties. However, we would not do that here
+    // as opertion mapping can support more complicated branching logic (right now we just assume
+    // it's always simple union), that Studio should not try to analyze.
+    skipMappingCheck: targetSetImpl instanceof OperationSetImplementation,
+    targetSetImpl,
+  };
+};
+
 export const getPropertyNodeMappingData = (
   graphManagerState: GraphManagerState,
   property: AbstractProperty,
   parentMappingData: QueryBuilderPropertyMappingData,
+  mapping: Mapping | undefined,
 ): QueryBuilderPropertyMappingData => {
   const parentTargetSetImpl = parentMappingData.targetSetImpl;
+  const superSetImplementations = getAllSuperSetImplementations(
```

Comment:
thinking about it why do you need mapping?
can't you use `parentTargetSetImpl.parent` as an input?